javalib/src/main/scala/java/net/URI.scala

```diff
@@ -113,6 +113,15 @@ final class URI(origStr: String) extends Serializable with Comparable[URI] {
       else cmpOpt(this._fragment, that._fragment)
     }
 
+    @inline def cmpPathQueryFragment(): Int = {
+      if (this._path != that._path)
+        cmpOpt(this._path, that._path)
+      else if (this._query != that._query)
+        cmpOpt(this._query, that._query)
+      else
+        cmpOpt(this._fragment, that._fragment)
+    }
+
```

Comment:
Same comment than on the previous commit. Probably easier to fix this in a more principled fashion.

Reply:
Refactored this same as above in https://github.com/scala-js/scala-js/commit/b2caa31c242ad85efee3fe748e9a7b925f220a0c